python/TestHarness/testers/PetscJacobianTester.py

```diff
@@ -26,6 +26,7 @@ def validParams():
                                           "at every non-linear iteration of every time step. This is only "
                                           "relevant for petsc versions >= 3.9.")
         params.addParam('turn_off_exodus_output', True, "Whether to set exodus=false in Outputs")
+        params.addParam('only_final_jacobian', False, "Check only final Jacobian comparison.")
```

Comment:
I hate to come in late and throw a bomb here (I was off all last week) but is this really what you want? The problem is that the jacobian of the initial iteration is wrong, not all iterations. My fear is that by the time you get to the last jacobian, either during the nonlinear iterations in a given timestep, or during multiple timesteps, you may not really testing a jacobian that is tightly coupled.